presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -3,27 +3,68 @@ package me.androidbox.presentation.agenda.viewmodel
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
 import me.androidbox.domain.agenda.usecase.UsersInitialsExtractionUseCase
+import me.androidbox.domain.authentication.ResponseState
 import me.androidbox.domain.authentication.preference.PreferenceRepository
+import me.androidbox.domain.authentication.remote.EventRepository
 import me.androidbox.presentation.agenda.screen.AgendaScreenEvent
 import me.androidbox.presentation.agenda.screen.AgendaScreenState
+import java.time.ZoneId
+import java.time.ZonedDateTime
 import javax.inject.Inject
 
 @HiltViewModel
 class AgendaViewModel @Inject constructor(
     private val preferenceRepository: PreferenceRepository,
-    private val usersInitialsExtractionUseCase: UsersInitialsExtractionUseCase
+    private val usersInitialsExtractionUseCase: UsersInitialsExtractionUseCase,
+    private val eventRepository: EventRepository
 ) : ViewModel() {
 
     private val _agendaScreenState = MutableStateFlow(AgendaScreenState())
     val agendaScreenState = _agendaScreenState.asStateFlow()
 
     init {
         getAuthenticatedUser()
+        fetchAgendaItems(ZonedDateTime.now())
+    }
+
+    private fun getStartOffCurrentDay(agendaDate: ZonedDateTime = ZonedDateTime.now(ZoneId.systemDefault())): ZonedDateTime {
+        return agendaDate
+            .toLocalDate()
+            .atStartOfDay(ZoneId.systemDefault())
```

Comment:
Didn't you created a function in your `DateTimeFormatterProvider` that already does this?